flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -2003,12 +2003,13 @@ class ScalarExprLowering {
 
       if (arg.passBy == PassBy::BaseAddress || arg.passBy == PassBy::BoxChar) {
         auto argAddr = [&]() -> ExtValue {
+          auto contiguous = Fortran::evaluate::IsSimplyContiguous(
```

Comment:
The `expr->Rank() > 0` below went along with the contiguity check, `IsSimplyContiguous` is meant to be called only on arrays.

Reply:
You could make `contiguous` be `const auto` to make it clear it is not a variable.

Reply:
> You could make `contiguous` be `const auto` to make it clear it is not a variable.

No longer having a contiguous variable, as I'm just relying on the one check to do the right thing.